simple.go

```diff
@@ -64,5 +66,51 @@ func GeneratePieceCIDFromFile(proofType abi.RegisteredProof, piece io.Reader, pi
 }
 
 func GenerateUnsealedCID(proofType abi.RegisteredProof, pieces []abi.PieceInfo) (cid.Cid, error) {
+	var sum abi.PaddedPieceSize
+	for _, p := range pieces {
+		sum += p.Size
+	}
+
+	ssize, err := SectorSizeForRegisteredProof(proofType)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	{
+		// pad remaining space with 0 CommPs
+		toFill := uint64(abi.PaddedPieceSize(ssize) - sum)
+		n := bits.OnesCount64(toFill)
+		for i := 0; i < n; i++ {
+			next := bits.TrailingZeros64(toFill)
+			psize := uint64(1) << next
+			toFill ^= psize
+
+			unpadded := abi.PaddedPieceSize(psize).Unpadded()
+			pieces = append(pieces, abi.PieceInfo{
+				Size:     unpadded.Padded(),
+				PieceCID: ZeroPieceCommitment(unpadded),
+			})
+		}
+	}
+
 	return ffi.GenerateUnsealedCID(proofType, pieces)
 }
+
+// TODO: remove this method after implementing it along side the registered proofs and importing it from there.
+func SectorSizeForRegisteredProof(p abi.RegisteredProof) (abi.SectorSize, error) {
+	switch p {
+	case abi.RegisteredProof_StackedDRG32GiBSeal:
```

Comment:
go does not have implicit fallthroughs, this code might not do what you expect it to.

Reply:
Thanks, fixed